### PR TITLE
Prevent additive effects of IorT when F is parallel. 

### PR DESCRIPTION
**Pull request by Baccata**
Fixes https://github.com/typelevel/cats/issues/3783

* Applies the same fix as https://github.com/typelevel/cats/pull/3777 
* Adds tests to verify non-additive behaviour of Left in  EitherT / IorT when F is Parallel 
* Adds law checks for additiveParallel on both IorT and EitherT

### CODE REVIEW

File: core/src/main/scala/cats/data/IorT.scala [hunk 1]
```diff
@@ -439,6 +439,60 @@ abstract private[data] class IorTInstances extends IorTInstances1 {
...
+   * (a, b).parTupled  // => IorT(IO(Chain(error1, error2).leftIor[Unit]))
+   * }}}
+   */
+  def accumulatingParallel[M[_], E](implicit
+    P: Parallel[M],
+    E: Semigroup[E]
+  ): Parallel.Aux[IorT[M, E, *], IorT[P.F, E, *]] { type Dummy } =
+    new Parallel[IorT[M, E, *]] {
+      type F[x] = IorT[P.F, E, x]
+      type Dummy // fix to make this one more specific than the catsDataParallelForIorTWithSequentialEffect, see https://github.com/typelevel/cats/pull/2335#issuecomment-408249775
```
Comment (djspiewak) [inline core/src/main/scala/cats/data/IorT.scala:494]: I think we can remove the `Dummy` types here, since they're just an implicit search workaround.

**Review by djspiewak — APPROVED** [review]
Looks good aside from my one minor comment.

File: core/src/main/scala/cats/data/IorT.scala (hunk 1 above)
Comment (Baccata) [reply core/src/main/scala/cats/data/IorT.scala]: Addressed 49fafb0

**Comment by larsrh** [comment]
I'm having trouble understanding the diff. It looks like there's just one additional non-implicit definition, no? In that case, why is this marked as source breaking?

**Comment by Baccata** [comment]
@larsrh I swapped to methods to help the diff. [Here's](https://github.com/typelevel/cats/pull/3791/files#diff-2bdfce46cbcc6ea0c7911c010533de85fe0e87a8ae7d4aaa854529d67a8b8369R504-R505) what's making it source-breaking.

**Review by larsrh — APPROVED** [review]

**Pull request merged**